unit_tests/src/test_replica.cpp

```diff
@@ -21,7 +21,7 @@ TEST_CASE("replica", "[replica]")
     load_client_api_plugins();
 
     auto conn_pool = irods::make_connection_pool();
-    auto conn = static_cast<rcComm_t>(conn_pool->get_connection());
+    auto conn = static_cast<rcComm_t&>(conn_pool->get_connection());
```

Comment:
This is actually a bug and needs to be adjusted.  With the current impl, the connection is being returned back to the pool, but it is still being used throughout the test.  This is fine because there is only one connection being used and it is synchronous, but its usage is incorrect.

Here is an alternative.
```c++
auto conn = conn_pool->get_connection();
rcComm_t& conn_ref = conn;
```